api/src/opentrons/protocol_engine/commands/heater_shaker/set_target_shake_speed.py

```
@@ -0,0 +1,59 @@
+"""Command models to shake the Heater Shaker Module."""
+from typing import Optional
+from typing_extensions import Literal, Type
+
+from pydantic import BaseModel, Field
+
+from ..command import AbstractCommandImpl, BaseCommand, BaseCommandCreate
+
+
+SetTargetShakeSpeedCommandType = Literal["heaterShakerModule/setTargetShakeSpeed"]
+
+
+class SetTargetShakeSpeedParams(BaseModel):
+    """Input parameters to start shaking a Heater Shaker Module."""
+
+    moduleId: str = Field(..., description="Unique ID of the Heater Shaker Module.")
+    # TODO(mc, 2022-02-24): for set temperature we use `temperature` (not `celsius`)
+    # but for shake we use `rpm` (not `speed`). This is inconsistent
+    rpm: int = Field(..., description="Target speed in rotations per minute.")
```

Typing this as an `int` *might* be confusing because JSON doesn't distinguish between ints and floats, so if you're reading the protocol schema, it looks like it's meaningful to pass something like `123.4`. But this will silently truncate it to `123`. (Pydantic....)

If the hardware needs an `int`, maybe we should accept a `float` here and it should be Protocol Engine's job to do the rounding?

Or maybe we don't care. I don't mind terribly either way.